lib/streamlit/elements/arrow.py

```diff
@@ -54,7 +55,7 @@ def _arrow_dataframe(
 
         Parameters
         ----------
-        data : pandas.DataFrame, pandas.Styler, pyarrow.Table, numpy.ndarray, Iterable, dict, or None
+        data : pandas.DataFrame, pandas.Styler, pyarrow.Table, numpy.ndarray, snowflake.snowpark.DataFrame, Iterable, dict, or None
```

Comment:
There is also the type definition above which we might want to add `snowflake.snowpark.DataFrame` as well, otherwise people will get typing warnings.

https://github.com/streamlit/streamlit/blob/71eb38ccec573c27811a18cd0b260d1a66bbffa9/lib/streamlit/elements/arrow.py#L41

Reply:
Also, the docstrings and types needs to be added to the dataframe selector methods as well.

Reply:
> There is also the type definition above which we might want to add snowflake.snowpark.DataFrame as well, otherwise people will get typing warnings.

I didn't add it to:
```Data = Union[DataFrame, Styler, pa.Table, ndarray, Iterable, Dict[str, List[Any]], None]```
because:
* Snowpark works only in Python 3.8 (we have Min Python Version tests and Max Python versions tests)
* We don't have Snowpark as Streamlit dependency and I guess we don't want to have it as a dep

If people will get warnings I guess we have to add `Any` to the line:
```Data = Union[DataFrame, Styler, pa.Table, ndarray, Iterable, Dict[str, List[Any]], Any, None]```

🤔 What can we do from this point?

Reply:
ahh I see 🤔 You can also add it as a string to the types which doesn't require any dependency:

```python
Data = Union[
    DataFrame,
    Styler,
    pa.Table,
    ndarray,
    Iterable,
    Dict[str, List[Any]],
    "snowflake.snowpark.DataFrame",
    None,
]
```

But I'm not 100% sure if that works, needs to be tested.

Reply:
alright, lets go with leaving that out for now